nginx.conf

```diff
@@ -41,5 +44,21 @@ http {
       default_type text/plain;
       return 200 "healthy\n";
     }
+
+    location /graphqlsha/ {
+        # Define caching rules
+        proxy_cache graphqlsha_requests;
+        proxy_cache_methods POST;  # Enable caching for POST requests
+        proxy_cache_key "$uri$request_body";  # Use URI and request body as cache key
+        proxy_cache_valid 200 ${CACHE_TTL};  # Cache 200 responses for 30 minutes
+        proxy_pass http://app;  # Backend server URL
+
+        proxy_buffers 8 8k;
+        proxy_buffer_size 16k; # required for large keys
```

Comment:
I'd make this configurable and provide defaults in the entrypoint script.

Reply:
I had a change of mind and decided to remove the proxy buffers for now to keep things simple. The caching change is big enough to go on it's own. I will add proxy_buffer later as needed ( or create a card )